### PR TITLE
Documentation of the current library code

### PR DESCRIPTION
**Pull request by davidcok**
By submitting this pull request, I confirm that my contribution is made under the terms of the MIT license.

### CODE REVIEW

File: src/dafny/DafnyCore.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+- [Dafny.Math](Math) --- basic mathematical operations and lemmas
+- [Dafny.Unicode](Unicode/Unicode) --- tools to manipulate unicode characters and strings
+- [Dafny.Collections](Collections/Collections) -- functions and lemmas about sequences, sets and maps
+- [Dafny.NonlinearArith](NonlinearArithmetic/NonlinearArithmetic) --  lemmas and functions for non-linear arithmetic
+- [Dafny.Relations](Relations) -- predicates and lemmas about properties of functions
+
+
+## How to use the library
+
+One uses the library on the command-line by including a path to the library in the `--library` option.
```
Comment (atomb) [inline src/dafny/DafnyCore.md:38]:
```suggestion
One uses the library on the command-line by including a path to the library as an argument to the `--library` option.
```

File: src/dafny/DafnyCore.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+At present, you must have a copy of the library in your file system to use the library. Either
+- obtain a release zip file from [the library github release page](https://github.com/dafny-lang/libraries/releases) and unzip it in a folder of your choice, or
```
Comment (atomb) [inline src/dafny/DafnyCore.md:41]: Do you have any thoughts about when/how we should start creating releases?

File: src/dafny/DafnyCore.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+- clone the library repository (`git clone https://github.com/dafny-lang/libraries.git`) to a location of your choice
+
+In either case, the relevant directory to point the `--library` option to is named `dafny`, either at the top of the zip file or at `libraries/src/dafny` in the clone, or to the `DafnyCore.dfy` file in that directory. 
+
+Some functionality is implemented using native target language capabilities (e.g. FileIO). In these cases, a compiled program must also includ the relevant target
+language files that are included in this Dafny Core distribution.
```
Comment (atomb) [inline src/dafny/DafnyCore.md:47]: Maybe refer to the `FileIO` docs here as an example?

File: src/dafny/FileIO/README.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
+# File I/O
```
Comment (atomb) [inline src/dafny/FileIO/README.md:1]: Maybe this file can go away now.

File: src/dafny/Functions.dfy-save [hunk 5]
```diff
@@ -0,0 +1,12 @@
+include "./Relations.dfy"
```
Comment (atomb) [inline src/dafny/Functions.dfy-save:1]: Did you mean to include this file?

File: src/dafny/NonlinearArithmetic/README.md [hunk 6]
```diff
@@ -0,0 +1,9 @@
+# Overview
```
Comment (atomb) [inline src/dafny/NonlinearArithmetic/README.md:1]: This text should presumably be migrated to `NonlinearArithmetic.md`.

**Review by atomb — CHANGES_REQUESTED** [review]

File: src/dafny/DafnyCore.md (hunk 2 above)
Comment (davidcok) [reply src/dafny/DafnyCore.md:41]: That will take some discussion with Robin, so I'm not taking bets.

File: src/dafny/FileIO/README.md (hunk 4 above)
Comment (davidcok) [reply src/dafny/FileIO/README.md:1]: Next round, if you don't mind.

File: src/dafny/DafnyCore.md (hunk 3 above)
Comment (davidcok) [reply src/dafny/DafnyCore.md:47]: Done

File: src/dafny/Functions.dfy-save (hunk 5 above)
Comment (davidcok) [reply src/dafny/Functions.dfy-save:1]: Nope

File: src/dafny/NonlinearArithmetic/README.md (hunk 6 above)
Comment (davidcok) [reply src/dafny/NonlinearArithmetic/README.md:1]: Guess it was easy enough to write that file.

**Review by atomb — APPROVED** [review]
Thanks for getting some documentation in place!

**Pull request closed without merging**